Fix empty reference location

kosack opened the pull request:
This adds a fallback dummy location to subarray.reference_location if the metadata in the simtel file doesn't exist.  The dummy location is on "null island" (lat=0, lon=0), but at the correct height as read from the simulation_config.


This was a problem for pre-Prod6 files, which didn't contain the metadata necessary to determine the reference location.

Tobychev commented:
Would it be possible to add a test that checks "proper" initialisation when metadata is missing?

maxnoe commented on ctapipe/io/simteleventsource.py:
```diff
@@ -513,15 +511,14 @@ def __init__(self, input_url=Undefined, config=None, parent=None, **kwargs):
-
-        self._subarray_info = self.prepare_subarray_info(
-            self.file_.telescope_descriptions, self.file_.header
-        )
```
Any particular reason for this move?

kosack replied:
Yes, it needs to come after parse_simulation_header, otherwise I can't use the SimulationConfigurationContainer

maxnoe commented:
Docs failure is fixed in main, please rebase

maxnoe commented:
One of the example notebooks no fails because it tries to print a subarray without a reference location:

```
WARNING: /home/runner/work/ctapipe/ctapipe/examples/core/InstrumentDescription.py failed to execute correctly: Traceback (most recent call last):
  File "/home/runner/work/ctapipe/ctapipe/examples/core/InstrumentDescription.py", line 49, in <module>
    newsub.info()
  File "/home/runner/work/ctapipe/ctapipe/ctapipe/instrument/subarray.py", line 129, in info
    printer(f"Height   : {self.reference_location.geodetic.height:.2f}")
AttributeError: 'NoneType' object has no attribute 'geodetic'
```

kosack commented:
> One of the example notebooks no fails because it tries to print a subarray without a reference location:

Seems it was `select_subarray()` not propegating the reference location. Fixed.  Also, I now changed SubarrayDescription's constructor to require the reference location (and telescope info), to avoid that it is ever set incorrectly.
